src/tap.cpp

```diff
@@ -41,27 +48,135 @@ TAP::TAP(QObject *parent) : QObject(parent) {
         close(mFd);
         exit(1);
     }
+    if (ioctl(mFd, TUNSETOFFLOAD, 0U) < 0) {
+        qCritical() << "Failed to set the TAP offload";
+    }
+
+    mIfaceName = ifr.ifr_name;
+    qDebug() << "TAP interface created:" << mIfaceName;
 
-    QString ifaceName = ifr.ifr_name;
-    qDebug() << "TAP interface created:" << ifaceName;
+    iffReset(244);
 
-    // Bring the interface up
-    if (QProcess::execute("ip", {"link", "set", "dev", ifaceName, "up"}) != 0) {
-        qCritical() << "Failed to bring up TAP interface";
+    char discard[1500];
+    auto drained = read(mFd, discard, 1500);
+    qDebug() << "Drained" << drained << "bytes";
+
+    // Poll the interface file descriptor
+    mNotifier = new QSocketNotifier(mFd, QSocketNotifier::Read, this);
+    QObject::connect(mNotifier, &QSocketNotifier::activated, this, &TAP::fdActivated);
+}
+
+void TAP::iffReset(int mtu) {
+    struct ifreq ifr = {};
+    auto sock = socket(PF_INET, SOCK_DGRAM, 0);
```

Comment:
Since this socket is just required to do ioctls, let's create it once as part of the constructor and keep the fd around.